File: backend/src/main/java/hu/szakdoga/backend/timetable/data/dto/SubjectDTO.java
package hu.szakdoga.backend.timetable.data.dto;

public class SubjectDTO {
    public final Long id;
    public final String name;
    public final String abbreviation;
    public final String color;
    public final String requirement;
    public final Long userId;

    public SubjectDTO(Long id, String name, String abbreviation, String color, String requirement, Long userId) {
        this.id = id;
        this.name = name;
        this.abbreviation = abbreviation;
        this.color = color;
        this.requirement = requirement;
        this.userId = userId;
    }
}

File: frontend/src/app/shared/model/timetable/dto/subject.dto.ts
export class SubjectDto {
  public readonly id: number | null;
  public readonly name: string;
  public readonly abbreviation: string;
  public readonly color: string;
  public readonly requirement: string;
  public readonly userId: number;

  constructor(
    name: string,
    abbreviation: string,
    color: string,
    requirement: string,
    userId: number,
    id: number | null = null,
  ){
    this.id = id;
    this.name = name;
    this.abbreviation = abbreviation;
    this.color = color;
    this.requirement = requirement;
    this.userId = userId;
  };
}
